modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPElementResourceImpl.java

```diff
@@ -305,6 +305,17 @@ public SXPElement putSXPElement(Long sxpElementId, SXPElement sxpElement)
 		sxpElement.setId(sxpElementId);
 
 		if (serviceBuilderSxpElement != null) {
+			if (serviceBuilderSxpElement.getReadOnly()) {
```

Comment:
I think this has a bit too many layers of 'if's. Maybe first something like:
```

if (serviceBuilderSxpElement == null) {
   return postSXPElement(sxpElement); 
}
...
```

And then getting more space by reversing the:

```
			if (serviceBuilderSxpElement.getReadOnly()) {
				return _sxpElementDTOConverter.toDTO(
					new DefaultDTOConverterContext(
						contextAcceptLanguage.isAcceptAllLanguages(),
						new HashMap<>(), _dtoConverterRegistry,
						contextHttpServletRequest,
						serviceBuilderSxpElement.getSXPElementId(),
						contextAcceptLanguage.getPreferredLocale(),
						contextUriInfo, contextUser),
					serviceBuilderSxpElement);
			}
```

to 

```
			if (!serviceBuilderSxpElement.getReadOnly()) {
		         	return _updateSxpElement(
				     serviceBuilderSxpElement.getSXPElementId(), sxpElement);
                        }

			return _sxpElementDTOConverter.toDTO(
					new DefaultDTOConverterContext(
						contextAcceptLanguage.isAcceptAllLanguages(),
						new HashMap<>(), _dtoConverterRegistry,
						contextHttpServletRequest,
						serviceBuilderSxpElement.getSXPElementId(),
						contextAcceptLanguage.getPreferredLocale(),
						contextUriInfo, contextUser),
					serviceBuilderSxpElement);
			}
```

Please check the PR if there are similar sections.